### ISSUE TITLE
Remove date format user setting

### ISSUE THREAD

**Issue by chenilim** [mattermost/focalboard#778]
Now that #715 will be fixed, let's use this ticket to track removing the Date format user setting.

Apologies for the round-about path this took. However, I believe the resulting design (MMMM DD, YYYY) is better overall in avoiding the need for a user setting in the first place.

**Comment by chenilim** [comment]
Hi @kamre, not sure why I can't assign this to you, but please use this ticket to submit the PR to remove the user setting. Thank you for all your help on this!

**Comment by kamre** [comment]
@chenilim @asaadmahmood Should we also change the date format for editing date in the input? Something unambiguous like `YYYY-MM-DD`

**Comment by asaadmahmood** [comment]
@kamre If possible, we should make it consistent to the MMMM DD, YYYY

**Comment by kamre** [comment]
@asaadmahmood I think there is no easy way to parse `MMMM DD, YYYY` back into `Date` because `Intl` doesn't have such functionality. And for different locales it looks really different:
![KanbanCardPreviewDateLocales](https://user-images.githubusercontent.com/5587620/126042900-6e762484-d922-435a-b052-99cfc403757b.gif)
Even editing this with full names of the months can lead to typos and screwing things up in my understanding.

**Comment by asaadmahmood** [comment]
@kamre Then we can just leave the format as it is for now, as the user is already selecting it by himself and the input probably also gives him a visual indication of the date selected.

**Comment by kamre** [comment]
@chenilim I have create a [branch for this issue](https://github.com/kamre/focalboard/compare/gh-715-consistent-date-display...kamre:gh-778-remove-preferred-date-format?expand=1) based on changes for #715

So after #748 is merged I will create new pull request.
